.github/workflows/ci.yml

```diff
@@ -0,0 +1,119 @@
+name: CI
+
+on:
+  pull_request:
+    branches: [ independent-releases ]
+  push:
+    branches: [ independent-releases ]
+  workflow_dispatch:
+
+jobs:
+  changes:
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: read
+    outputs:
+      packages: ${{ steps.filter.outputs.changes }}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: dorny/paths-filter@v3
+        id: filter
+        with:
+          filters: |
+            scan-docker-image: 'security-actions/scan-docker-image/**'
+            sca: 'security-actions/sca/**'
+            scan-rust: 'security-actions/scan-rust/**'
+            semgrep: 'security-actions/semgrep/**'
+            sign-docker-image: 'security-actions/sign-docker-image/**'
+            code-build-actions: 'code-build-actions/build-js-sdk/**'
+            lua-lint: 'code-check-actions/lua-lint/**'
+            rust-lint: 'code-check-actions/rust-lint/**'
+            slack-actions: 'slack-actions/workflow-notification/**'
+            cleanup: 'pr-previews/cleanup/**'
+            up-to-date: 'pr-previews/up-to-date/**'
+            validate: 'pr-previews/validate/**'
+
+  setup-dependencies:
+    needs: changes
+    runs-on: ubuntu-latest
+    if: ${{ needs.changes.outputs.packages != '[]' }}
+    outputs:
+      nodeVersion: ${{ steps.node-version.outputs.nodeVersion }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Setup Node
```

Comment:
Refer pnpm global package caching at https://github.com/actions/setup-node/blob/main/docs/advanced-usage.md#caching-packages-data 

setup-pnpm and setup-node with cache input `pnpm` and `cache-depdency-path` should be used which by default caches `~/.pnpm-store` instead of `node_modules`